dvc/output/base.py

```diff
@@ -429,16 +436,14 @@ def get_used_cache(self, **kwargs):
                     )
                 )
             logger.warning(msg)
-            return NamedCache()
+            return None, NamedCache()
```

Comment:
Maybe it would make sense to push `dir` into NamedCache? So that NamedCache could contain both dirs and files(e.g. `cache["local"].dirs)? Just an idea, not sure if it would be nicer looking in the end.

Reply:
NamedCache supports nesting cache items (for `.dir` checksums) after 02aaf42, the end result is a bit cleaner now